R/read.magpie.R

```diff
@@ -130,11 +131,31 @@ read.magpie <- function(file_name, file_folder = "", file_type = NULL, as.array
       getCells(readMagpie) <- sub("_", ".", getCells(readMagpie))
     }
     attr(readMagpie, "comment") <- .readComment(fileName, commentChar = comment.char)
-  } else if (fileType == "cs4" | fileType == "cs4r") {
+  } else if (fileType %in% c("cs4", "cs4r")) {
     x <- read.csv(fileName, comment.char = comment.char, header = FALSE,
                   check.names = check.names, stringsAsFactors = TRUE)
     readMagpie <- as.magpie(x, tidy = TRUE)
     attr(readMagpie, "comment") <- .readComment(fileName, commentChar = comment.char)
+  } else if (fileType == "cs5") {
+    .metaExtract <- function(fileName, commentChar) {
+      comment <- .readComment(fileName, commentChar = commentChar)
+      m <- grep("^META ", comment)
+      metadata <- comment[m]
+      comment <- comment[-m]
+      mNames <- sub("^META (.*):.*$", "\\1", metadata)
```

Comment:
This breaks when `:` is in one of the names, e.g. `*META names: x, y, coun:try, cell`. This is admittedly very unlikely, but it's easy to fix:
```suggestion
      mNames <- sub("^META ([^:]*):.*$", "\\1", metadata)
```

Reply:
fixed